### ISSUE TITLE
unknown error while dumping

### ISSUE THREAD

**Issue by ConnorRigby** [ConnorRigby/rx8-ecu-dump#1]
i've pulled the firmware from a few ECUs now, and every once in a while i've received the following error, or something similar. retrying usually solves it, but i did get it twice in a row at one point. It's worth noting that this has only happened to me on Windows. I have not experienced this on Linux

```
[ECUDump] j2534 connection initialized ok
[ECUDump] Got VIN = JM1FE173370212600
[ECUDump] Got calibration ID = N3M5EF00013H6020
[ECUDump] Removing old ROM JM1FE173370212600-N3M5EF00013H6020.bin
[ECUDump] Diag sesion initialized ok
[ECUDump] Got seed = { C8, 5B, 2B }
[ECUDump] Got key  = { 16, 5A, 72 }
[ECUDump] Unlocked ECU
[ECUDump] Starting ROM dump, this will take a moment..
[RX8] [readMem] unknown error     ( 57%)
[348937463]00 00 07 E8 7F 23 78
[RX8] [readMem] Unknown message
[349032408]63 8F 0C F5 30 61 DC E0 03 31 03 8D 07 00 09 60 EC 88 01 8F 03 00 09 E3 01 A0 02 24 30 E1 00 24 10 DA 2A E4 03 9B 41 65 DC DC 26 35 43 8F 09 00 09 D3 27 62 30 22 28 8F 04 00 09 60 EC 88 01 8D 0D 00 09 D2 24 63 20 63 3C 33 43 8D 0D 00 09 35 43 8F 0A 00 09 60 EC 88 01 8F 06 00 09 4A 0B 00 09 FB 0A E2 00 A0 15 2C 21 61 C1 D2 1B 63 21 31 36 8D 09 00 09 4A 0B 00 09 FB 0A E5 01 D2 17 42 0B 64 C1 A0 06 2C 01 D3 16 F5 B8 F3 38 4A 0B F5 30 FB 0A D2 0F D3 0F 22 E0 23 D0 4F 26 6A F6 6B F6 6C F6 6D F6 00 0B 6E F6 80 A4 CB 74 CB 5E CB 6E B2 AC CB 78 00 07 28 58 00 04 03 C6 00 07 CA 08 00 00 FF FF FF FF CB 7A 00 07 CA 12 00 00 23 F4 FF FF CB 7D FF FF CB 7C 00 07 CA 10 00 00 24 60 00 07 CA 0C 92 65 E3 01 91 64 22 30 F4 18 D5 35 F5 4A D3 35 64 30 62 4C 42 08 32 5C 65 23 60 4C 40 15 8F 05 75 04 74 FF 63 4C 43 15 8D FB F5 4B 00 0B 00 09 94 4F D3 2D 43 2B E5 00 D4 2C D3 2B 43 2B E5 00 94 47 D3 29 43 2B E5 01 94 44 D3 27 43 2B E5 00 2F E6 4F 22 DE 26 D4 25 4E 0B 00 09 94 39 4E 0B 00 09 94 37 4E 0B 00 09 4F 26 00 0B 6E F6 2F E6 2F D6 2F C6 2F B6 4F 22 93 2D 64 30 9B 2C 6D 4C 9E 2B 60 D3 88 01 8F 07 EC 00 E5 01 D1 19 41 0B 64 B1 2B 01 A0 06 2E C1 2B C1 E5 01 D2 15 42 0B 64 E1 2E 01 94 1A 2D D8 8F 0A 00 09 93 15 62 31 D0 11 61 01 32 12 8F 03 00 09 E3 01 A0 01 24 30 24 C0 4F 26 6B F6 6C F6 6D F6 00 0B 6E F6 CB 92 B4 C0 87 A0 87 A2 AA CC CB 80 CB 82 CB 86 FF FF FF FF CB 9C 00 07 28 18 00 04 04 14 FF FF 87 A4 00 04 04 BA 00 00 24 60 00 07 CA 1C 4F 22 D2 4B 63 20 23 38 8F 18 00 09 91 7F 60 10 60 0C 88 01 8D 06 00 09 93 7A 60 30 60 0C 88 01 8F 0C 00 09 93 75 60 30 60 0C 88 01 8D 13 00 09 92 70 60 20 60 0C 88 01 8D 0D 00 09 D2 3D 63 21 D0 3D 61 01 33 12 8D 0A 00 09 93 64 62 01 61 31 31 22 8F 04 00 09 D4 39 D3 39 43 0B E5 01 93 5A D1 34 62 31 4F 26 00 0B 21 21 2F E6 2F D6 2F C6 2F B6 2F A6 2F 96 2F 86 FF FB FF EB FF DB FF CB 90 4A 4F 22 93 49 7F E8 92 48 FE 08 FD 38 FF 28 91 45 60 10 DE 2B 80 F8 94 42 4E 0B E5 00 2F 00
[ECUDump] Failed to dump remainder of ROM 49980
```

**Comment by ConnorRigby** [comment]
happened again on windows
```
PS C:\Users\konno\Desktop\rx8\ecu\ecudump\Debug> .\ecudump.exe
[ECUDump] j2534 connection initialized ok
[ECUDump] Got VIN = SE3P116167
[ECUDump] Got calibration ID = N3YMEE00013H60A0
[ECUDump] Diag sesion initialized ok
[ECUDump] Got seed = { 33, 53, 70 }
[ECUDump] Got key  = { 0B, 6C, 5E }
[ECUDump] Unlocked ECU
[ECUDump] Starting ROM dump, this will take a moment..
[RX8] [readMem] unknown error     ( 13%)
[31989137]00 00 07 E8 7F 23 78
[RX8] [readMem] Unknown message
[32083119]63 00 05 A7 6E 00 05 74 B2 00 05 CD BC 00 05 43 38 00 05 DD E0 00 06 19 90 00 06 3B 60 00 05 FB 50 00 06 09 22 00 06 4D AC 00 05 FB C2 00 06 00 CA 00 06 81 DC 00 06 82 CC 00 06 75 44 00 06 73 6C 00 02 5F D8 00 02 F0 B6 00 02 D7 9C 00 03 97 98 00 02 19 88 00 01 42 34 00 02 73 C4 00 01 68 64 00 02 3B 60 00 02 FB 6C 00 02 FC B0 00 03 03 98 00 03 26 5C 00 03 21 7C 00 01 7C D8 00 02 27 90 00 02 27 54 00 02 23 6C 00 03 55 9C 00 03 55 C8 00 02 2E 54 00 02 31 44 00 02 43 58 00 02 93 24 00 03 56 9C 00 05 F2 74 00 05 F2 F0 00 05 F3 A0 00 05 F2 4C 00 02 AD 78 00 03 F8 58 00 02 61 50 00 01 6A 8C 00 05 99 22 00 05 97 70 00 05 DB 60 00 05 DC 1C 00 05 DC 76 00 05 90 F0 00 05 95 F6 00 05 8C 48 00 05 8A 28 00 05 78 3C 00 05 79 D4 00 05 7C 9C 00 05 7D BE 00 05 7E FC 00 05 80 44 00 05 81 80 00 05 82 B8 00 05 83 DE 00 02 3B 92 00 02 20 C8 00 01 47 78 00 05 9A 34 4F 22 D3 8C 43 0B 00 09 D2 8B 42 0B 00 09 D3 8B 43 0B 00 09 D2 8A 42 0B 00 09 D3 8A 43 0B 00 09 D2 89 42 0B 00 09 D3 89 43 0B 00 09 D2 88 42 0B 00 09 D3 88 43 0B 00 09 D2 87 42 0B 00 09 D3 87 43 0B 00 09 D2 86 42 0B 00 09 D3 86 43 0B 00 09 D2 85 42 0B 00 09 D3 85 43 0B 00 09 D2 84 42 0B 00 09 D3 84 43 0B 00 09 D2 83 42 0B 00 09 D3 83 43 0B 00 09 D2 82 42 0B 00 09 D3 82 43 0B 00 09 D2 81 42 0B 00 09 D3 81 43 0B 00 09 D2 80 42 0B 00 09 D3 80 43 0B 00 09 D2 7F 42 0B 00 09 D3 7F 43 0B 00 09 D2 7E 42 0B 00 09 D3 7E 43 0B 00 09 D2 7D 42 0B 00 09 D3 7D 43 0B 00 09 D2 7C 42 0B 00 09 D3 7C 43 0B 00 09 D2 7B 42 0B 00 09 D3 7B 43 0B 00 09 D2 7A 42 2B 4F 26 2F E6 4F 22 D3 79 43 0B EE 00 60 0E 20 08 8D 01 00 09 EE 01 64 E3 4F 26 D3 75 43 2B 6E F6 4F 22 D3 74 43 0B 00 09 D2 73 42 0B 00 09 D3 73 43 0B 00 09 D2 72 42 0B 00 09 D3 72 43 2B 4F 26 4F 22 D3 71 43 0B 00 09 D2 70 42 0B 00 09 D3 70 43 0B 00 09 D2 6F 42 0B 00 09 D3 6F 43 0B 00 09 D2 6E 42 0B 00 09 D3 6E 43 0B 00 09 D2 6D 42 0B 00 09 D3 6D 43 0B 00 09 D2 6C 42 0B 00 09 D3 6C 43 2B 4F 26 4F 22 D3 6B 43 0B 00 09
[ECUDump] Failed to dump remainder of ROM 11300
```

**Comment by ConnorRigby** [comment]
Fixed by [4f4baaa](https://github.com/ConnorRigby/rx8-ecu-dump/commit/4f4baaa12caefd310f26a22799491b16f8c62bc0)
